Due to rather strict compiler setting, building Pi-hole FTL with GCC16
fails with:

build/pihole-ftl-6.7/src/config/validator.c: In function ‘sanitize_dns_hosts’:
build/pihole-ftl-6.7/src/config/validator.c:653:13: error: variable ‘i’ set but not used [-Werror=unused-but-set-variable=]
  653 |         int i = 0;
      |             ^

Add a patch containing the corresponding commit from the development branch.

Upstream: https://github.com/pi-hole/FTL/commit/0b82796
Signed-off-by: Andreas Ziegler <15275159+aeolio@users.noreply.github.com>
---
From 0b827963d5366703279f6bd3fd54d2b43361a50b Mon Sep 17 00:00:00 2001
From: darkexplosiveqwx <101737077+darkexplosiveqwx@users.noreply.github.com>
Date: Mon, 6 Jul 2026 23:32:12 +0200
Subject: [PATCH] fix build error from leftover variable

Signed-off-by: darkexplosiveqwx <101737077+darkexplosiveqwx@users.noreply.github.com>
---
 src/config/validator.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/config/validator.c b/src/config/validator.c
index 29c42bb57..64ad3217d 100644
--- a/src/config/validator.c
+++ b/src/config/validator.c
@@ -650,8 +650,7 @@ void sanitize_dns_hosts(union conf_value *val)
 	// Walk the linked list directly: cJSON_GetArrayItem() is O(index) and
 	// cJSON_GetArraySize() in the loop condition re-walks the whole list
 	// every iteration, which made this loop O(n^2). item->next is O(1).
-	int i = 0;
-	for(cJSON *item = val->json != NULL ? val->json->child : NULL; item != NULL; item = item->next, i++)
+	for(cJSON *item = val->json != NULL ? val->json->child : NULL; item != NULL; item = item->next)
 	{
 
 		// Check if it's a string
